script/utils/sim-sequence.sh

```diff
@@ -0,0 +1,308 @@
+  exit 1
+
+}
+
+
+#TODO: GET THE ADDRESSES FROM SUPERCHAIN-REGISTRY IN THE FUTURE, since they are the safes addresses this sholdn't change so often so this fine.
+## ETHEREUM
+Security_Council_Safe=0xc2819DC788505Aac350142A7A707BF9D03E3Bd03
+Foundation_Upgrade_Safe=0x847B5c174615B1B7fDF770882256e2D3E95b9D92
+Foundation_Operation_Safe=0x9BA6e03D8B90dE867373Db8cF1A58d2F7F006b3A
+Proxy_Admin_Owner_Safe=0x5a0Aae59D09fccBdDb6C6CcEB07B7279367C3d2A
+DESTROY_ANIVIL_AFTER_EXECUTION=true
+##############################################################
+# Simulates a sequence of tasks for a given network by running them against an Anvil
+# fork with state overrides disabled.
+#
+# Usage:
+#   ./script/utils/sim-sequence.sh <network> <array-of-task-IDs>
+#
+# Example:
+#   ./script/utils/sim-sequence.sh eth "021 022 base-003 ink-001"
+
+# TODO start an anvil fork and run all simulations against the same node.
+# TODO verify the following test cases, push a commit verifying each one has the expected result in CI:
+#   - expected to pass: ./script/utils/sim-sequence.sh eth "021 022 base-003 ink-001"
+#   - expected to pass: ./script/utils/sim-sequence.sh eth "021 022 base-003"
+#   - expected to fail: ./script/utils/sim-sequence.sh eth "021 base-003 ink-001 022"
+#   - expected to fail: ./script/utils/sim-sequence.sh eth "021 base-003 ink-001"
+#   - expected to fail: ./script/utils/sim-sequence.sh eth "021 025"
+#   - expected to fail: create a dir called 021-tmp and run ./script/utils/sim-sequence.sh eth "021 022 base-003 ink-001"
+set -euo pipefail
+
+# --- Functions ---
+
+error_exit() {
+  echo "Error: $1" >&2
+  exit 1
+}
+
+# Disable state overrides in Simulation.sol. The only state override currently
+# disabled is the nonce override.
+disable_state_overrides() {
+  # Create a backup
+  local simulation_file="$root_dir/lib/base-contracts/script/universal/Simulation.sol"
+  local backup_file="${simulation_file}.bak"
+  cp "$simulation_file" "$backup_file"
+
+  # Comment out lines containing 'state = addNonceOverride'.
+  awk '/state = addNonceOverride/ { print "        // " $0; next } { print }' "$simulation_file" > "${simulation_file}.tmp"
+
+  # Replace the original file with the modified version, and ensure cleanup on exit.
+  mv "${simulation_file}.tmp" "$simulation_file"
+  trap cleanup EXIT
+}
+
+cleanup() {
+  unset SIMULATE_WITHOUT_LEDGER
+
+  local simulation_file="$root_dir/lib/base-contracts/script/universal/Simulation.sol"
+  local backup_file="${simulation_file}.bak"
+
+  if [[ -f "$backup_file" ]]; then
+    mv "$backup_file" "$simulation_file"
+  fi
+
+  # Kill the anvil fork at the end if it was started by this script
+  if $DESTROY_ANIVIL_AFTER_EXECUTION; then
+    echo "Anvil is still open. To kill it, please use the command below:"
+    echo "ps aux | grep anvil | grep -v grep | awk '{print \$3}' | xargs kill"
+    #TODO: In the future, we should kill the anvil fork here?
```

Comment:
Still a TODO it seem? Feel free to create tracking issues for any of these future improvements

Reply:
Done here -> https://github.com/ethereum-optimism/superchain-ops/issues/646